node-graph/gpu-executor/src/lib.rs

```diff
@@ -176,6 +176,10 @@ pub enum ShaderInput<E: GpuExecutor + ?Sized> {
 	ReadBackBuffer(E::BufferHandle, Type),
 }
 
+// TODO: is this safe??
+unsafe impl<E: GpuExecutor + ?Sized> Send for ShaderInput<E> {}
+unsafe impl<E: GpuExecutor + ?Sized> Sync for ShaderInput<E> {}
```

Comment:
In this case we would need to use a where bound requiring that all of the types which could be used in the enum (`BufferHandle`, `TextureView`, `TextureHandle`) are Send+Sync otherwise this is definitely unsound